.evergreen/generate_evergreen_tasks.js

```diff
@@ -461,6 +461,54 @@ for (const {
   }
 }
 
+// Running CSFLE tests with mongocryptd
+const MONGOCRYPTD_CSFLE_TASKS = MONGODB_VERSIONS
+  .filter((mongoVersion) => { // Only run on server versions >= 4.2
+    const numericVersion = Number(mongoVersion);
+    return !Number.isNaN(numericVersion) && numericVersion >= 4.2;
+  })
```

Comment:
We actually have a package called `semver` that we use for version comparison:

also, I think we should test on latest / rapid as well

```suggestion
  .filter((version) => ['latest', 'rapid'].includes(version) || semver.gte(version, '4.2'))
```